client/etcd/client.go

```diff
@@ -152,19 +152,24 @@ func (c *csclient) createTxnStore(opts kv.Options) (kv.TxnStore, error) {
 	if err := opts.Validate(); err != nil {
 		return nil, err
 	}
-	return c.txnGen(c.opts.Zone(), c.cacheFileFn(), opts.Namespace(), opts.Environment())
+	return c.txnGen(c.opts.Zone(), c.cacheFileFn(), opts.Logger(), opts.Namespace(), opts.Environment())
 }
 
 func (c *csclient) kvGen(fn cacheFileForZoneFn) etcdsd.KVGen {
 	return etcdsd.KVGen(func(zone string) (kv.Store, error) {
-		return c.txnGen(zone, fn)
+		return c.txnGen(zone, fn, c.logger)
 	})
 }
 
-func (c *csclient) newkvOptions(zone string, cacheFileFn cacheFileForZoneFn, namespaces ...string) etcdkv.Options {
+func (c *csclient) newkvOptions(
+	zone string,
+	cacheFileFn cacheFileForZoneFn,
+	logger log.Logger,
+	namespaces ...string,
+) etcdkv.Options {
 	opts := etcdkv.NewOptions().
 		SetInstrumentsOptions(instrument.NewOptions().
-			SetLogger(c.logger).
+			SetLogger(logger).
```

Comment:
nit: would this mean that you have to explicitly set the logger in the options to receive updates? Maybe we should check if it's `nil` and fall back to `c.logger` if so?

Reply:
If the logger is not set, it falls back to using the client's logger. The logger set in the options is more like an override.